src/app/components/media/MediaTypeDisplayIcon.js

```diff
@@ -11,6 +11,42 @@ import Public from '../../icons/public.svg';
 import Twitter from '../../icons/twitter.svg';
 import YouTube from '../../icons/youtube.svg';
 
+export function mediaTypeFromUrl(url) {
+  let mediaType = 'Link';
+
+  if (/^(https?:\/\/)?((www\.)?youtube\.com|youtu\.be)\/.+$/.test(url)) {
+    mediaType = 'Youtube';
+  } else if (/^(https?:\/\/)?((www\.)?instagram\.com)\/.+$/.test(url)) {
+    mediaType = 'Instagram';
+  } else if (/^(https?:\/\/)?((www\.)?twitter\.com)\/.+$/.test(url)) {
```

Comment:
We should also test for `x.com` here and resolve it to Twitter since those URLs are starting to get embedded in the wild

Reply:
Great call, I'm adding x.com, thanks!